### ISSUE TITLE
代码申请

### ISSUE THREAD

**Issue by jiduo1767** [ZJier/DKDMN#3]
作者您好，我想复现您的代码，但是没有找到pca降维和数据集拼接部分的代码，自己写之后有一些错误无法排除，能否分享一下这部分的代码？感谢您！

**Comment by ZJier** [comment]
虽然我不清楚你提到的错误是什么，但是我建议你先运行train_unet.py文件，然后运行feature_extract_unet.py文件。
你提到的数据集拼接部分的代码并不是独立的，它就是feature_extract_unet.py文件的inference_by_t函数。当你运行train_unet.py获得权重文件后，直接运行feature_extract_unet.py文件，inference_by_t函数会帮你保存两个特征文件，其中path = "%s/t%s_%s_full.pkl" % (save_feature_path_prefix, ti, index)就是完整的分布知识，不用拼接(如果你获得了这个分布知识，你可以直接np.load读取，就可以发现它的空间维度和原始数据是一致的，但通道维度不是，这里需要你直接降维)。其次，PCA降维直接针对获得的完整分布知识文件执行就可以，这个容易进行。

**Comment by jiduo1767** [comment]
感谢您的解答，那就是把运行完train_unet.py和feature_extract_unet.py之后保存的文件进行降维后在[DKDMN.py]文件中替换为input运行[DKDMN.py]就可以吗？（或者我直接用您提供的百度网盘中的Houston_Knowledge_K30.pkl.npy作为输入也可以对吗）感谢！

> 虽然我不清楚你提到的错误是什么，但是我建议你先运行train_unet.py文件，然后运行feature_extract_unet.py文件。 你提到的数据集拼接部分的代码并不是独立的，它就是feature_extract_unet.py文件的inference_by_t函数。当你运行train_unet.py获得权重文件后，直接运行feature_extract_unet.py文件，inference_by_t函数会帮你保存两个特征文件，其中path = “%s/t%s_%s_full.pkl” % （save_feature_path_prefix， ti， index）就是完整的分布知识，不用拼接（如果你获得了这个分布知识，你可以直接np.load读取，就可以发现它的空间维度和原始数据是一致的，但通道维度不是，这里需要你直接降维）。其次，PCA降维直接针对获得的完整分布知识文件执行就可以，这个容易进行。

**Comment by ZJier** [comment]
> 感谢您的解答，那就是把运行完train_unet.py和feature_extract_unet.py之后保存的文件进行降维后在[DKDMN.py]文件中替换为input运行[DKDMN.py]就可以吗？（或者我直接用您提供的百度网盘中的Houston_Knowledge_K30.pkl.npy作为输入也可以对吗）感谢！
> 
> > 虽然我不清楚你提到的错误是什么，但是我建议你先运行train_unet.py文件，然后运行feature_extract_unet.py文件。 你提到的数据集拼接部分的代码并不是独立的，它就是feature_extract_unet.py文件的inference_by_t函数。当你运行train_unet.py获得权重文件后，直接运行feature_extract_unet.py文件，inference_by_t函数会帮你保存两个特征文件，其中path = “%s/t%s_%s_full.pkl” % （save_feature_path_prefix， ti， index）就是完整的分布知识，不用拼接（如果你获得了这个分布知识，你可以直接np.load读取，就可以发现它的空间维度和原始数据是一致的，但通道维度不是，这里需要你直接降维）。其次，PCA降维直接针对获得的完整分布知识文件执行就可以，这个容易进行。

是这样的。不过，不是必须要替换掉input，你可以把“知识”作为输入的一部分，例如torch.cat([ori_data, hsi_know], axis=2)。或者，你希望用“知识”执行其他操作，都没有问题。你可以直接用百度网盘中的数据，但是我建议你能把它跑通，这样有利于你做其他数据集。感谢您对我们工作的关注！
